### DISCUSSION TITLE
Study Level Reporting - Assessment Standardization

### DISCUSSION THREAD

**Discussion by jwildfire** [Gilead-Public/gsm#256] [Ideas]
Going to write up a few posts on this topic including some thoughts on "Assessment Standardization", "Input Data Generation", "Raw data checking" and "Study reporting"  

First up is: 

# Assessment Standardization
- `Assess` functions should always return a list. 
- Remove `bDataList` option  and always return full `lAssess` object. Users can always subset the output or call individual pieces of the assessment as needed. 
- Create `lAssess$function`, which indicates which assessment pipeline was run to create the assessment (AE_Assess has `class` = "AE")
- Create `lAssess$params` which captures the input parameters sent to the assessment. 
- Add a `bViz` option to all `Assess` functions. When `true` return the visualization (and any associated objects) as part of the list
- Still need to work out a classification schema (#20). We could just remove `cLabel` and `cAssessment`  and allow users to pass in a named vector defining the hierarchy (`vClass`?).  Would give lots of flexibility: 
```
# All of these would be fine for the same study
vClass<-c(Study="ABC123", Class="Disposition", Type="Study Drug Discontinuation", Reason="Lost to Follow-up")
vClass<-c(Study="ABC123", Class="AE", Type="AE")
vClass<-c(Study="ABC123", Class="AE", Type="SAE")
```

**Comment by gwu05** [comment]
@jwildfire In general, agree with the standardization recommendations. 

For the vClass, okay with the flexibility, but unclear on how the labeling will work (i.e., in the Study Report that summarizes all the assessments, what are the labels that go in to that report like Risk Category, Risk Assessment, and Risk Label)?

Do we want to enable dynamic interaction with visuals and assessments? Since we've moved away from the Shiny tool, and plan to have users not decide thresholds directly, then our current structure should work. If we need to allow users to tweak a diagnostic plot, which then change the boundaries/flags on the assessment plot, then grouping everything into the assess structure will not be useful (although, since we have the individual pieces, we can still create that functionality easily). On those lines, we still need to add a diagnostic plot for each assessment, as that is an essential piece to decide on an appropriate threshold.

**Reply by jwildfire** [reply]
> For the vClass, okay with the flexibility, but unclear on how the labeling will work (i.e., in the Study Report that summarizes all the assessments, what are the labels that go in to that report like Risk Category, Risk Assessment, and Risk Label)?

Yeah, the benefit of this approach is that we can define different hierarchies for different use cases. The drawback is that we have to sort that out and standardize for any different use cases. I'm not opposed to something more prescriptive (e.g. strCategory, strAssessment, strLabel) I'm just a bit worried that it will break down as we keep expanding the number of assessments and scenarios. 
 
> Do we want to enable dynamic interaction with visuals and assessments? Since we've moved away from the Shiny tool, and plan to have users not decide thresholds directly, then our current structure should work. If we need to allow users to tweak a diagnostic plot, which then change the boundaries/flags on the assessment plot, then grouping everything into the assess structure will not be useful (although, since we have the individual pieces, we can still create that functionality easily). 

Agreed, when we introduce more robust interactivity (like interactive threshold tweaks), it will be by calling the workflow piecemeal instead of wrapped up in the assessment framework. 

> On those lines, we still need to add a diagnostic plot for each assessment, as that is an essential piece to decide on an appropriate threshold.

I assume we'll want diagnostic metric/plot per analysis function? Might be worth starting a separate discussion covering this and documenting what you're thinking for each assessment.
